src/ILLink.CodeFix/DAMCodeFixProvider.cs

```diff
@@ -0,0 +1,114 @@
+			return diagDescriptorsArrayBuilder.ToImmutable ();
+		}
+
+		public static ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => GetSupportedDiagnostics ();
+
+		public sealed override ImmutableArray<string> FixableDiagnosticIds => SupportedDiagnostics.Select (dd => dd.Id).ToImmutableArray ();
+
+		private protected static LocalizableString CodeFixTitle => new LocalizableResourceString (nameof (Resources.DynamicallyAccessedMembersCodeFixTitle), Resources.ResourceManager, typeof (Resources));
+
+		private protected static string FullyQualifiedAttributeName => DynamicallyAccessedMembersAnalyzer.FullyQualifiedDynamicallyAccessedMembersAttribute;
+
+		protected static SyntaxNode[] GetAttributeArguments (ISymbol targetSymbol, SyntaxGenerator syntaxGenerator, Diagnostic diagnostic)
+		{
+			object id = Enum.Parse (typeof (DiagnosticId), diagnostic.Id.Substring (2));
+			switch (id) {
+			case DiagnosticId.DynamicallyAccessedMembersMismatchParameterTargetsThisParameter:
+				return new[] { syntaxGenerator.AttributeArgument (syntaxGenerator.TypedConstantExpression (targetSymbol.GetAttributes ().First (attr => attr.AttributeClass?.ToDisplayString () == DynamicallyAccessedMembersAnalyzer.FullyQualifiedDynamicallyAccessedMembersAttribute).ConstructorArguments[0])) };
+			case DiagnosticId.DynamicallyAccessedMembersMismatchFieldTargetsThisParameter:
+				return new[] { syntaxGenerator.AttributeArgument (syntaxGenerator.TypedConstantExpression (targetSymbol.GetAttributes ().First (attr => attr.AttributeClass?.ToDisplayString () == DynamicallyAccessedMembersAnalyzer.FullyQualifiedDynamicallyAccessedMembersAttribute).ConstructorArguments[0])) };
+			case DiagnosticId.DynamicallyAccessedMembersMismatchParameterTargetsMethodReturnType:
+				return new[] { syntaxGenerator.AttributeArgument (syntaxGenerator.TypedConstantExpression (targetSymbol.GetAttributes ().First (attr => attr.AttributeClass?.ToDisplayString () == DynamicallyAccessedMembersAnalyzer.FullyQualifiedDynamicallyAccessedMembersAttribute).ConstructorArguments[0])) };
+			default:
+				return Array.Empty<SyntaxNode> ();
+			}
+		}
+
+		public sealed override FixAllProvider GetFixAllProvider ()
+		{
+			// See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+			return WellKnownFixAllProviders.BatchFixer;
+		}
+
+		public override async Task RegisterCodeFixesAsync (CodeFixContext context)
+		{
+			var document = context.Document;
+			var root = await document.GetSyntaxRootAsync (context.CancellationToken).ConfigureAwait (false);
+			var diagnostic = context.Diagnostics.First ();
+			var props = diagnostic.Properties;
+			var model = await document.GetSemanticModelAsync (context.CancellationToken).ConfigureAwait (false);
+			SyntaxNode diagnosticNode = root!.FindNode (diagnostic.Location.SourceSpan);
```

Comment:
Should I add a test case associated with this fix?

Reply:
Yup, that's exactly what my change fixes. I would agree -- creating a test case, confirming it throws an exception, then applying the fix and confirming it no longer does so is a good idea.

Reply:
Ah, actually it looks like `diagnosticNode` is actually properly checked below, s oit won't throw an exception. Probably just won't work (the type will be `ArgumentSyntax` instead of `InvocationExpressionSyntax`).